presto-native-execution/presto_cpp/main/functions/remote/client/tests/RemoteFunctionRestTest.cpp

```diff
@@ -0,0 +1,331 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdio>
+
+#include <boost/asio.hpp>
+#include <folly/init/Init.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "presto_cpp/main/functions/remote/client/RestRemoteClient.h"
+#include "presto_cpp/main/functions/remote/client/RestRemoteFunction.h"
+#include "presto_cpp/main/functions/remote/client/tests/rest/RemoteFunctionRestService.h"
+#include "presto_cpp/main/functions/remote/client/tests/rest/examples/RemoteDoubleDivHandler.h"
+#include "presto_cpp/main/functions/remote/client/tests/rest/examples/RemoteFibonacciHandler.h"
+#include "presto_cpp/main/functions/remote/client/tests/rest/examples/RemoteInverseCdfHandler.h"
+#include "presto_cpp/main/functions/remote/client/tests/rest/examples/RemoteRemoveCharHandler.h"
+#include "presto_cpp/main/functions/remote/client/tests/rest/examples/RemoteStrLenHandler.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/exec/tests/utils/PortUtil.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+#include "velox/type/fbhive/HiveTypeParser.h"
+
+using ::facebook::velox::test::assertEqualVectors;
+using namespace facebook::velox;
+namespace facebook::presto::functions::rest::test {
+namespace {
+
+class RemoteFunctionRestTest
+    : public velox::functions::test::FunctionBaseTest,
+      public testing::WithParamInterface<velox::functions::remote::PageFormat> {
+ public:
+  void SetUp() override {
+    auto servicePort = exec::test::getFreePort();
+    location_ = fmt::format(kHostAddressTemplate_, servicePort);
+    auto wrongServicePort = exec::test::getFreePort();
+    wrongLocation_ = fmt::format(kHostAddressTemplate_, wrongServicePort);
+
+    // Create shared RestRemoteClient instances
+    remoteClient_ = std::make_shared<RestRemoteClient>(location_);
+    wrongRemoteClient_ = std::make_shared<RestRemoteClient>(wrongLocation_);
+
+    initializeServer(servicePort);
+    registerRemoteFunctions();
+  }
+
+  ~RemoteFunctionRestTest() override {
+    if (serverThread_ && serverThread_->joinable()) {
+      ioc_.stop();
+      serverThread_->join();
+    }
+  }
+
+ private:
+  template <typename Handler>
+  void registerRemoteFunctionHelper(
```

Comment:
Add a comment indicating this is registering the server and client handlers for the remote functions.